source/Irrlicht/CGLTFMeshFileLoader.cpp

```diff
@@ -81,13 +80,24 @@ bool CGLTFMeshFileLoader::isALoadableFileExtension(
 */
 IAnimatedMesh* CGLTFMeshFileLoader::createMesh(io::IReadFile* file)
 {
-	tinygltf::Model model {};
+	if (file->getSize() <= 0) {
+		return nullptr;
+	}
+	std::optional<tiniergltf::GlTF> model = tryParseGLTF(file);
+	if (!model.has_value()) {
+		return nullptr;
+	}
 
-	if (file->getSize() <= 0 || !tryParseGLTF(file, model)) {
+	if (!(model->buffers.has_value()
+			&& model->bufferViews.has_value()
+			&& model->accessors.has_value()
+			&& model->meshes.has_value()
+			&& model->nodes.has_value())) {
+		throw std::runtime_error("what");

```

Comment:
sfan5 will be overjoyed to see you remembered not to leave the error message empty.

Reply:
oops, I forgot to remove this :sweat_smile: